core/src/main/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystem.java

```diff
@@ -231,15 +230,19 @@ public OutputStream appendFile(S3Object f) throws FileSystemException, IOExcepti
 	@Override
 	public Message readFile(S3Object f) throws FileSystemException, IOException {
 		try {
-			final S3Object file = s3Client.getObject(bucketName, f.getKey());
-			final S3ObjectInputStream is = file.getObjectContent();
-
-			return new Message(is);
+			return new S3Message(s3Client.getObject(bucketName, f.getKey()));
 		} catch (AmazonServiceException e) {
 			throw new FileSystemException(e);
 		}
 	}
 
+	private class S3Message extends Message {
+		
+		public S3Message(S3Object file) {
+			super(() -> file.getObjectContent(), null, file.getClass());
+		}
```

Comment:
Hier zou je ook `getContentLength()` kunnen gebruiken om de size te bepalen

Reply:
Die zit niet in de SDK of API

Reply:
Hij zit in de `ObjectMetadata` maar laten we dat maar voor een andere keer laten...